### PR TITLE
feature/radar

### PR DESCRIPTION
**Pull request by ria0308**

### CODE REVIEW

File: src/components/common/ui/radar/utils/drawAngelLines.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const getAngelLines = (values, data) => {
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/drawAngelLines.ts:1]: add types to props

File: src/components/common/ui/radar/utils/drawAngelLines.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  for (let i = 0; i < values.length; i++) {
+    const arr = new Array(values.length).fill(0);
+    arr[i] = 10;
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/drawAngelLines.ts:4]: what is that?

File: src/components/common/ui/radar/utils/drawAngelLines.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    data.push({
+      label: 'Data',
+      data: arr,
+      borderWidth: 1,
+      backgroundColor: '#404040',
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/drawAngelLines.ts:9]: can we get those coolers from theme?

File: src/components/common/ui/radar/utils/drawAngelLines.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+      borderColor: '#404040',
+      pointBorderColor: 'grey.black',
+      pointBorderWidth: 0,
+      pointBackgroundColor: 'transparent',
+      pointRadius: 0,
+      lineTension: 0.35,
+      color: 'red',
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/drawAngelLines.ts:16]: again color from theme

File: src/components/common/ui/radar/utils/drawAngelLines.ts [hunk 5]
```diff
@@ -0,0 +1,21 @@
+const getAngelLines = (values, data) => {
+  for (let i = 0; i < values.length; i++) {
+    const arr = new Array(values.length).fill(0);
+    arr[i] = 10;
+    data.push({
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/drawAngelLines.ts:5]: mutating prop inside function???

File: src/components/common/ui/radar/Radar.tsx [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+  Filler,
+  Legend,
+  LinearScale,
+  LineElement,
+  PointElement,
+  RadialLinearScale,
+  Tooltip,
+} from 'chart.js';
+
+ChartJS.register(
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/Radar.tsx:15]: this and props interface should be under the imports

File: src/components/common/ui/radar/Radar.tsx [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+interface RadarProps {
+  values: Array<number>;
+}
+import getAngelLines from '@/components/common/ui/radar/utils/drawAngelLines';
+import getData from '@/components/common/ui/radar/utils/drawData';
+import getBackground from '@/components/common/ui/radar/utils/fillBackground';
+import getOptions from '@/components/common/ui/radar/utils/getOptions';
+
+import * as styles from './Radar.styles';
+const RadarChart: FC<RadarProps> = ({ values }) => {
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/Radar.tsx:33]: add empty line

File: src/components/common/ui/radar/utils/drawData.ts [hunk 8]
```diff
@@ -0,0 +1,26 @@
+const getData = values => {
+  const names = [];
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/drawData.ts:2]: why do we need that?

File: src/components/common/ui/radar/utils/drawData.ts [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+  }
+  return {
+    labels: names,
+    color: 'red',
+    datasets: [
+      {
+        color: 'white',
+        label: 'Data',
+        data: values,
+        backgroundColor: 'rgba(188, 61, 61, 0.54)',
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/drawData.ts:14]: can we accomplish that with mui theme? it has `alpha` function for it

File: src/components/pages/test-pages/radar-test-page/RadarTestPage.tsx [hunk 10]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
+import { Box } from '@mui/material';
+
+import RadarChart from '@/components/common/ui/radar';
+
+import * as styles from './RadarTestPage.styles';
+const RadarTestPage = () => {
```
Comment (andrewvitrenko) [inline src/components/pages/test-pages/radar-test-page/RadarTestPage.tsx:7]: add empty line

File: src/components/pages/test-pages/radar-test-page/RadarTestPage.tsx [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+  return (
+    <Box sx={styles.wrapper}>
+      <RadarChart values={[9, 8, 6, 3, 9.9, 9, 5, 6, 1, 3, 9.9, 4, 6]} />
```
Comment (andrewvitrenko) [inline src/components/pages/test-pages/radar-test-page/RadarTestPage.tsx:10]: move to const above component

File: src/components/common/ui/radar/Radar.styles.ts [hunk 12]
```diff
@@ -0,0 +1,6 @@
+import { SxProps, Theme } from '@mui/material/styles';
+
+export const background: SxProps<Theme> = {
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/Radar.styles.ts:3]: rename to `container`

File: src/components/common/ui/radar/utils/fillBackground.ts [hunk 13]
```diff
@@ -0,0 +1,20 @@
+const getBackground = (values, data) => {
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/fillBackground.ts:1]: rename function

File: src/components/common/ui/radar/utils/fillBackground.ts [hunk 14]
```diff
@@ -0,0 +1,20 @@
...
+  for (let i = 0; i < 11; i++) {
+    const arr = new Array(values.length).fill(i);
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/fillBackground.ts:3]: 😭

File: src/components/common/ui/radar/utils/fillBackground.ts [hunk 15]
```diff
@@ -0,0 +1,20 @@
...
+    data.push({
+      label: 'Data',
+      data: arr,
+      borderWidth: 1,
+      backgroundColor: i % 2 == 0 ? '#212121' : '#242424',
+      borderColor: '#353535',
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/fillBackground.ts:9]: same about theme color

File: src/components/common/ui/radar/Radar.tsx [hunk 16]
```diff
@@ -0,0 +1,45 @@
...
+  const data = getData(values);
+  data.datasets = getAngelLines(values, data.datasets);
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/Radar.tsx:35]: we should change that

File: src/components/pages/test-pages/radar-test-page/RadarTestPage.styles.ts [hunk 17]
```diff
@@ -0,0 +1,21 @@
...
+  justifyContent: 'center',
+  padding: '36px',
+  gap: '18px',
+  ul: {
+    display: 'flex',
+    flexDirection: 'column',
+    gap: '8px',
+  },
+  backgroundColor: {
+    mobile: '#212121',
```
Comment (andrewvitrenko) [inline src/components/pages/test-pages/radar-test-page/RadarTestPage.styles.ts:16]: colors from theme

File: src/components/common/ui/radar/Radar.styles.ts [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+import findImage from '@/components/common/ui/radar/utils/findImage';
+
+export const background = (lables): SxProps<Theme> => ({
+  width: '100%',
+  aspectRatio: {
+    mobile: '1/1',
+    desktopSemiMedium: '1200/742',
+  },
+  backgroundImage: {
+    mobile: findImage(lables, 'mobile'),
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/Radar.styles.ts:12]: rename to `getImage`

File: src/components/common/ui/radar/Radar.tsx [hunk 19]
```diff
@@ -0,0 +1,55 @@
...
+ChartJS.register(
+  RadialLinearScale,
+  LinearScale,
+  PointElement,
+  LineElement,
+  Filler,
+  Tooltip,
+  Legend,
+);
+interface RadarProps {
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/Radar.tsx:24]: move interface under imports

File: src/components/common/ui/radar/utils/findImage.ts [hunk 20]
```diff
@@ -0,0 +1,41 @@
...
+  '11-mobile': 'url("/assets/radar-background/mobile-11.png")',
+  '13-mobile': 'url("/assets/radar-background/mobile-13.png")',
+  '15-mobile': 'url("/assets/radar-background/mobile-15.png")',
+};
+const keyPhrases = {
+  laborant: 'Відповідність лабораторних',
+  practician: 'Відповідність практичних',
+};
+
+const getBackgroundImage = (labels, screen = 'desktop') => {
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/findImage.ts:16]: add types in props

**Review by andrewvitrenko — CHANGES_REQUESTED** [review]

File: src/components/common/ui/radar/utils/findImage.tsx [hunk 21]
```diff
@@ -0,0 +1,47 @@
+import { Mobile11 } from '@/components/common/ui/radar/utils/radar-background/11-mobile';
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/utils/findImage.tsx:1]: use dynamic imports for those elements

**Review by andrewvitrenko — CHANGES_REQUESTED** [review]

File: src/components/common/ui/radar/Radar.tsx [hunk 22]
```diff
@@ -0,0 +1,56 @@
...
+const RadarChart: FC<RadarProps> = ({ info }) => {
+  const labels = info.map(a => a.name);
+  const data = getData(info.map(({ mark }) => mark));
+  const options = getOptions();
+  const isMobile = useMediaQuery(theme.breakpoints.down('desktopSemiMedium'));
+  return (
+    <Box sx={styles.background(isMobile)}>
+      <Radar options={options} data={data} />
+      {getBackgroundImage(labels, isMobile, [
+        'Lecturer',
```
Comment (andrewvitrenko) [inline src/components/common/ui/radar/Radar.tsx:48]: get this data from props from page. contact `@DButov` for info about types

**Review by andrewvitrenko — CHANGES_REQUESTED** [review]

File: src/components/common/ui/radar/Radar.tsx (hunk 22 above)
Comment (Nanaier) [reply src/components/common/ui/radar/Radar.tsx:49]: will be done later by ```@sOleksandr```

File: src/components/pages/test-pages/radar-test-page/RadarTestPage.styles.ts (hunk 17 above)
Comment (Nanaier) [reply src/components/pages/test-pages/radar-test-page/RadarTestPage.styles.ts:16]: will be done later, when we have the palette

**Review by OleksandrHumeniuk — APPROVED** [review]

**Review by andrewvitrenko — APPROVED** [review]

**Pull request merged**